src/StoreApi/Schemas/ProductSchema.php

```diff
@@ -478,13 +524,18 @@ protected function get_term_list( \WC_Product $product, $taxonomy = '' ) {
 			return [];
 		}
 
-		$return = [];
+		$return           = [];
+		$default_category = (int) get_option( 'default_product_cat', 0 );
 
 		foreach ( $terms as $term ) {
 			$link = get_term_link( $term, $taxonomy );
 
 			if ( is_wp_error( $link ) ) {
-				$link = false;
+				continue;
+			}
+
+			if ( $term->term_id === $default_category ) {
```

Comment:
Found a bug whilst testing where the 'uncategorized' category was being listed. This prevents that happening.